### PR TITLE
chore(v3): fix e2e flow for create, add resource/capability and provision remote

### PR DESCRIPTION
**Pull request by jayzhang**

### CODE REVIEW

**Review by HuihuiWu-Microsoft — DISMISSED** [review]

**Comment by jayzhang** [comment]
This PR fix and refactor some contract while e2e integration test for "create a new project", "add resource",  "add capability", "provision remote env" flow.

**Review by XiaofuHuang — APPROVED** [review]
Approve APIM changes

File: packages/fx-core/src/plugins/resource/aad/v3/index.ts [hunk 1]
```diff
@@ -139,10 +179,8 @@ export class AadAppForTeamsPluginV3 implements v3.FeaturePlugin {
     });
 
     //init aad part in local settings or env state
-    if (!envInfo.state[BuiltInFeaturePluginNames.aad]) {
-      envInfo.state[BuiltInFeaturePluginNames.aad] = {
-        secretFields: ["clientSecret"],
-      };
+    if (!envInfo.state[BuiltInFeaturePluginNames.aad].secretFields) {
```
Comment (KennethBWSong) [inline packages/fx-core/src/plugins/resource/aad/v3/index.ts:182]: Will `envInfo.state[BuiltInFeaturePluginNames.aad]` be empty?

Comment (jayzhang) [reply packages/fx-core/src/plugins/resource/aad/v3/index.ts:182]: No, solution will guarantee the existence of the corresponding configs

**Review by KennethBWSong — APPROVED** [review]

**Review by 1yefuwang1 — APPROVED** [review]

**Review by nliu-ms — APPROVED** [review]

File: packages/fx-core/src/plugins/resource/bot/v3/index.ts [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+  CheckThrowSomethingMissing,
+  PackDirectoryExistenceError,
+  PreconditionError,
+  TemplateZipFallbackError,
+  UnzipError,
+} from "./error";
+import { ensureSolutionSettings } from "../../../solution/fx-solution/utils/solutionSettingsHelper";
+
+@Service(BuiltInFeaturePluginNames.bot)
+export class NodeJSBotPluginV3 implements v3.FeaturePlugin {
```
[the remark this answers is not in the document]
Comment (jayzhang) [reply packages/fx-core/src/plugins/resource/bot/v3/index.ts:80]: it includes add capability scenario, so commit together

**Review by hund030 — APPROVED** [review]

**Review by xzf0587 — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Pull request closed without merging**